### PR TITLE
Handle perpetually firing KubeVersionMismatch when using Amazon EKS

### LINKED ISSUE #144: KubeVersionMismatch on Amazon EKS
Hi,

I found the following github issue, but noticed a corresponding issue hadn't been opened here and I am running into the same problem. https://github.com/coreos/prometheus-operator/issues/1977

`kubernetes_build_info`
```
kubernetes_build_info{buildDate="2018-12-06T01:35:29Z",compiler="gc",endpoint="https-metrics",gitCommit="753b2dbc622f5cc417845f0ff8a77f539a4213ea",gitTreeState="clean",gitVersion="v1.11.5",goVersion="go1.10.3",instance="10.12.11.142:10250",job="kubelet",major="1",minor="11",namespace="kube-system",node="ip-10-86-10-142.us-west-2.compute.internal",platform="linux/amd64",service="prometheus-operator-kubelet"}	1

kubernetes_build_info{buildDate="2018-12-06T01:35:29Z",compiler="gc",endpoint="https-metrics",gitCommit="753b2dbc622f5cc417845f0ff8a77f539a4213ea",gitTreeState="clean",gitVersion="v1.11.5",goVersion="go1.10.3",instance="10.12.13.100:10250",job="kubelet",major="1",minor="11",namespace="kube-system",node="ip-10-86-12-100.us-west-2.compute.internal",platform="linux/amd64",service="prometheus-operator-kubelet"}	1

kubernetes_build_info{buildDate="2018-12-06T01:35:29Z",compiler="gc",endpoint="https-metrics",gitCommit="753b2dbc622f5cc417845f0ff8a77f539a4213ea",gitTreeState="clean",gitVersion="v1.11.5",goVersion="go1.10.3",instance="10.12.12.127:10250",job="kubelet",major="1",minor="11",namespace="kube-system",node="ip-10-86-14-127.us-west-2.compute.internal",platform="linux/amd64",service="prometheus-operator-kubelet"}	1

kubernetes_build_info{buildDate="2018-12-06T23:13:14Z",compiler="gc",endpoint="https",gitCommit="6bad6d9c768dc0864dab48a11653aa53b5a47043",gitTreeState="clean",gitVersion="v1.11.5-eks-6bad6d",goVersion="go1.10.3",instance="10.12.55.98:443",job="apiserver",major="1",minor="11+",namespace="default",platform="linux/amd64",service="kubernetes"}	1

kubernetes_build_info{buildDate="2018-12-06T23:13:14Z",compiler="gc",endpoint="https",gitCommit="6bad6d9c768dc0864dab48a11653aa53b5a47043",gitTreeState="clean",gitVersion="v1.11.5-eks-6bad6d",goVersion="go1.10.3",instance="10.12.55.131:443",job="apiserver",major="1",minor="11+",namespace="default",platform="linux/amd64",service="kubernetes"}	1
```

More clearly, the `gitVersion`s don't match
`sum by (gitVersion) (kubernetes_build_info)`
```
{gitVersion="v1.11.5"}	3
{gitVersion="v1.11.5-eks-6bad6d"}	2
```

### PR DESCRIPTION
**Pull request by pgporada**
Fixes #144

### CODE REVIEW

File: alerts/system_alerts.libsonnet [hunk 1]
```diff
@@ -20,7 +20,7 @@
           {
             alert: 'KubeVersionMismatch',
             expr: |||
-              count(count(kubernetes_build_info{%(notKubeDnsSelector)s}) by (gitVersion)) > 1
+              count(count by (gitVersion) (label_replace(kubernetes_build_info{%(notKubeDnsSelector)s},"gitVersion","$1","gitVersion","(.*)(-eks.*)"))) > 1
```
[the remark this answers is not in the document]
Comment (pgporada) [reply alerts/system_alerts.libsonnet]: Sure, would you want to see `(.*)(-(.*))` ?

**Comment by pgporada** [comment]
Is there anything else that you need me to do with this before it can be merged in?

**Comment by pgporada** [comment]
Thank you for the detailed breakdown. I appreciate that. I like the idea of explicitly matching the semver.

**Comment by metalmatze** [comment]
Especially matching the semver directly, is a way better approach!

**Pull request merged**